pkg/operator/ceph/object/objectstore.go

```diff
@@ -661,44 +672,52 @@ func CreatePools(context *Context, clusterSpec *cephv1.ClusterSpec, metadataPool
 	return nil
 }
 
-func createSimilarPools(context *Context, pools []string, clusterSpec *cephv1.ClusterSpec, poolSpec cephv1.PoolSpec, pgCount, ecProfileName string) error {
+func createSimilarPools(ctx *Context, pools []string, clusterSpec *cephv1.ClusterSpec, poolSpec cephv1.PoolSpec, pgCount, ecProfileName string) error {
+	waitGroup, _ := errgroup.WithContext(context.TODO())
+
 	for _, pool := range pools {
-		// create the pool if it doesn't exist yet
-		name := poolName(context.Name, pool)
-		if poolDetails, err := ceph.GetPoolDetails(context.Context, context.clusterInfo, name); err != nil {
-			// If the ceph config has an EC profile, an EC pool must be created. Otherwise, it's necessary
-			// to create a replicated pool.
-			var err error
-			if poolSpec.IsErasureCoded() {
-				// An EC pool backing an object store does not need to enable EC overwrites, so the pool is
-				// created with that property disabled to avoid unnecessary performance impact.
-				err = ceph.CreateECPoolForApp(context.Context, context.clusterInfo, name, ecProfileName, poolSpec, pgCount, AppName, false /* enableECOverwrite */)
+		// Avoid the loop re-using the same value with a closure
+		pool := pool
+		waitGroup.Go(func() error {
```

Comment:
>> How about factoring this to a separate function? Maybe it's just personal preference for code readability, but I think it would be helpful.

> ​Not sure what you want to factor here, can you be more specific?

For  example: 
```go
...
waitGroup.Go(createRGWPool)
...
func createRGWPool() error {
  // move inline func implementation here
}
```

Reply:
Oh I see, looks more readable although we have plenty of function closures elsewhere in the code already ^^

Reply:
The short function closures are fine, it's the longer function closures that are less readable IMO, but we can leave it this way.